### DISCUSSION TITLE
unRAID backup solution

### DISCUSSION THREAD

**Discussion by Sptzzz** [immich-app/immich#8482] [Q&A]
Hi all, first time Immich user and I'm in absolute awe at how good it is!! Can't even believe I can finally de-g00gle myself.

I have specifically 2 questions if I may.

1) Backups. 
I'm running the app Backup/Restore Appdata. I've set it so that it also backs up the external volume for Immich. So that both the database and images data is in sync. Is this sound for a complete backup?

2) Restoring.
This solution has always worked for me for every single container I run + the data I choose to backup. My question is related to the above. As I'm planning on changing my build as I need to expand my storage and would like to know if the above method will work well on doing a restore when I finish the changes I need to make. (new ssds, re-creating cache pool etc)

**Comment by mmomjian** [comment]
I don't exactly know how Backup/Restore appdata works, but filesystem-level backups of Postgres would need to be done while the database is shut down. If this is not the case, you need to use `pg_dump` - you can start here: https://immich.app/docs/administration/backup-and-restore

**Comment by thoroughmas** [comment]
I use the Appdata Backup plugin to automatically stop my Immich & Postgres containers and back up their contents. Separately I'm backing up my photos library/profile/upload folders. Is this sufficient, or do I need to look at doing that database dump as well?

**Reply by Sptzzz** [reply]
This is exactly what I'm doing. Also backing up the media folder for Immich so the database never gets mismatched. Never attempted a restore so fingers crossed.

**Comment by mmomjian** [comment]
If the containers are stopped during the backup that should be safe.

**Reply by thoroughmas** [reply]
Thanks.
Hopefully at some point I'll get around to testing the backups to be totally sure.

**Comment by DrSpaldo** [comment]
I would also be interested in finding the outcome of this one, as to the best method within Unraid to backup & restore

**Comment by LackesLab** [comment]
I am also running unraid and had issues restoring the backed up postgres data. I hardly recommend (as others suggested already) : Use pgdump sperately to dump the database. You can use the UserScripts Plugin and the following scripts:
```
#!/bin/bash

# Check if array is started
ls /mnt/disk[1-9]* 1>/dev/null 2>/dev/null
if [ $? -ne 0 ]
then
   echo "ERROR:  Array must be started before using this script"
   exit
fi


BACKUP_DIR="/mnt/user/Backups/PostgresDatabases/"

databases=`docker exec postgres psql -U postgres -l -t | cut -d'|' -f1 | sed -e 's/ //g' -e '/^$/d'`


for db in $databases; do  if [ "$db" != "postgres" ] && [ "$db" != "template0" ] && [ "$db" != "template1" ] && [ "$db" != "template_postgis" ]; then    
    FOLDER=$BACKUP_DIR/$db

    echo "Backing up $db"
    mkdir -p "$FOLDER"

    FILENAME=${db}_$(date +%Y-%m-%d-%H.%M.%S).sql
    FILEPATH=$FOLDER/$FILENAME
    docker exec postgres pg_dump --clean -h localhost -U postgres -d $db > $FILEPATH
    tar czf $FILEPATH.tgz -C $FOLDER $FILENAME
    rm $FILEPATH

    fi
done

find $BACKUP_DIR/ -type f -name '*.tgz' -mtime +7 -exec rm {} \;
```

**Comment by zaggash** [comment]
Dbbackup app is doing a great job at it also
